src/components/graph/graph.helper.js

```diff
@@ -43,18 +44,36 @@ const NODE_PROPS_WHITELIST = ['id', 'highlighted', 'x', 'y', 'index', 'vy', 'vx'
  * @param  {number} width - the width of the container area of the graph.
  * @param  {number} height - the height of the container area of the graph.
  * @param  {number} gravity - the force strength applied to the graph.
+ * @param  {callback} onCollide - the method takes a node as its param and returns a collision force radius.
  * @returns {Object} returns the simulation instance to be consumed.
  * @memberof Graph/helper
  */
-function _createForceSimulation(width, height, gravity) {
+function _createForceSimulation(width, height, gravity, onCollide) {
     const frx = d3ForceX(width / 2).strength(CONST.FORCE_X);
     const fry = d3ForceY(height / 2).strength(CONST.FORCE_Y);
     const forceStrength = gravity;
+    const onCollision = _isValidCollisionFunction(onCollide) ? onCollide : d3ForceCollide;
```

Comment:
I don't think I'm really understanding this ternary here. So if the client does not specifies an `onCollision` function we will pass `d3ForceCollide` as that is going to be called inside itself?

IMO we should have collision active only if the client specifies the `onCollision` function:

```javascript
if (_isValidCollisionFunction(onCollide)) {
return d3ForceSimulation()
        .force('charge', d3ForceManyBody().strength(forceStrength))
        .force('x', frx)
        .force('y', fry)
        .force('y', fry)
        .force('collide', onCollision);
} else {
return d3ForceSimulation()
        .force('charge', d3ForceManyBody().strength(forceStrength))
        .force('x', frx)
        .force('y', fry)
        .force('y', fry);
}
```

The client could then pass something like:

```javascript
...
const onCollision = d3.forceCollide().radius(function(d) { return d.r + 0.5; }).iterations(2));
...
```

See https://bl.ocks.org/mbostock/31ce330646fa8bcb7289ff3b97aab3f5